---
title: "class19"
author: "Colin Mach"
format: pdf
---

```{r echo = FALSE}
cdc <- data.frame(
                                 Year = c(1922L,1923L,1924L,1925L,
                                          1926L,1927L,1928L,1929L,1930L,1931L,
                                          1932L,1933L,1934L,1935L,1936L,
                                          1937L,1938L,1939L,1940L,1941L,1942L,
                                          1943L,1944L,1945L,1946L,1947L,
                                          1948L,1949L,1950L,1951L,1952L,
                                          1953L,1954L,1955L,1956L,1957L,1958L,
                                          1959L,1960L,1961L,1962L,1963L,
                                          1964L,1965L,1966L,1967L,1968L,1969L,
                                          1970L,1971L,1972L,1973L,1974L,
                                          1975L,1976L,1977L,1978L,1979L,1980L,
                                          1981L,1982L,1983L,1984L,1985L,
                                          1986L,1987L,1988L,1989L,1990L,
                                          1991L,1992L,1993L,1994L,1995L,1996L,
                                          1997L,1998L,1999L,2000L,2001L,
                                          2002L,2003L,2004L,2005L,2006L,2007L,
                                          2008L,2009L,2010L,2011L,2012L,
                                          2013L,2014L,2015L,2016L,2017L,2018L,
                                          2019L),
         No..Reported.Pertussis.Cases = c(107473,164191,165418,152003,
                                          202210,181411,161799,197371,
                                          166914,172559,215343,179135,265269,
                                          180518,147237,214652,227319,103188,
                                          183866,222202,191383,191890,109873,
                                          133792,109860,156517,74715,69479,
                                          120718,68687,45030,37129,60886,
                                          62786,31732,28295,32148,40005,
                                          14809,11468,17749,17135,13005,6799,
                                          7717,9718,4810,3285,4249,3036,
                                          3287,1759,2402,1738,1010,2177,2063,
                                          1623,1730,1248,1895,2463,2276,
                                          3589,4195,2823,3450,4157,4570,
                                          2719,4083,6586,4617,5137,7796,6564,
                                          7405,7298,7867,7580,9771,11647,
                                          25827,25616,15632,10454,13278,
                                          16858,27550,18719,48277,28639,32971,
                                          20762,17972,18975,15609,18617)
       )
```

> Q1.) Plot of case numbers of persussis over time.

```{r}
library(ggplot2)
baseplot <- ggplot(cdc) + aes(x = Year, y = No..Reported.Pertussis.Cases) + geom_line() + geom_point() + labs(title = "Cases of Pertussis in US 1922-2019")
baseplot
```

> Q2.)

```{r}
baseplot + geom_vline(xintercept = 1946, color = "blue") + geom_vline(xintercept = 1996, color = "red")
```

> Q3.) What happened during these time periods?

There was a resurgence after the switch to the aP vaccine since the vaccine may be less efective than the wP vaccine and may not provide as much immunity since the pertussis may have evolved and adapted from the aP vaccine. The spike was also correlated with vaccine hesistancy since that was when people were misinformed that vaccines cause autism which has had a lasting impact since the 2000s.

# CMI-PB

```{r}
library(jsonlite)
subject <- read_json("https://www.cmi-pb.org/api/subject", simplifyVector = TRUE) 
```

> Q4.) Number of infancy vaccinated subjects in the dataset?

```{r}
head(subject, 3)
table(subject$infancy_vac)
```

There are 96 vaccinated subjects

> Q5.) How many Male and Female subjects are in the dataset

```{r}
table(subject$biological_sex)
```
There are 66 female and 30 male subjects

> Q6.) What is the breakdown of race and biological sex

```{r}
table(subject$biological_sex, subject$race)
```
> Q7.) average age of wP individuals and average age of aP individuals and are they siginificantly different

```{r}
library(lubridate)
subject$age <- today() - ymd(subject$year_of_birth)
library(dplyr)

ap <- subject %>% filter(subject$infancy_vac == "aP")

round(summary(time_length(ap$age, "years")))
```

```{r}
wp <- subject %>% filter(infancy_vac == "wP")
round(summary(time_length(wp$age, "years")))
```

```{r}
int <- ymd(subject$date_of_boost) - ymd(subject$year_of_birth)
age_at_boost <- time_length(int, "year")
head(age_at_boost)
```

> Q9.) Are these two groups statistically different in terms of average age

```{r}
ggplot(subject) +
  aes(time_length(age, "year"),
      fill=as.factor(infancy_vac)) +
  geom_histogram(show.legend=FALSE) +
  facet_wrap(vars(infancy_vac), nrow=2)
```
Graph shows that they are very signficantly different since one is skewed to the left while the wP is more spread out

```{r}
specimen <- read_json("https://www.cmi-pb.org/api/specimen", simplifyVector = TRUE)
titer <- read_json("https://www.cmi-pb.org/api/ab_titer", simplifyVector = TRUE)
```

> Q9.) 

```{r}
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```

> Q10.) 

```{r}
abdata <- inner_join(titer, meta)
dim(abdata)
```

> Q11.) 

```{r}
table(abdata$isotype)
```

> Q12.)

```{r}
table(abdata$visit)
```

```{r}
ig1 <- abdata %>% filter(isotype == "IgG1", visit!=8)
head(ig1)
```

> Q13.)

```{r}
ggplot(ig1) +
  aes(MFI, antigen) +
  geom_boxplot() +
  facet_wrap(vars(visit), nrow = 2)
```
> Q14.) 

FIM 2/3 show differences in the level of igG1 antibody titers recognizing them over time and FHA average increases over time.

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(visit), nrow=2) +
  theme_bw()
```

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot(show.legend = FALSE) + 
  facet_wrap(vars(infancy_vac, visit), nrow=2)
```

> Q15.) 

```{r}
filter(ig1, antigen=="Measles") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = FALSE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```
```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot(show.legend = FALSE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```
> Q16.) 

FIM 2/3 increases over time and then peaks at visit 5 and then declines after.

> Q17.)

aP seems to have a higher detection antigen levels after visit 4 than wP which is more pronounced as visits increase.

```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url, simplifyVector = TRUE) 
ssrna <- inner_join(rna, meta)
```

> Q18.)

```{r}
ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)
```

> Q19.)

Expression peaks around Visit 4.

> Q20.)

No this does not match the titer data since antigen has a long shelf life meaning that continuous gene expression is not needed in order to provide the necessary amount of antigen levels as they also peak at different times and that gene expression leads to the antigen so it makes sense that expression would peak before antigen.

```{r}
ggplot(ssrna) +
  aes(tpm, col=infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit))
```

```{r}
ssrna %>%  
  filter(visit==4) %>% 
  ggplot() +
    aes(tpm, col=infancy_vac) + geom_density() + 
    geom_rug() 
```

